### PR TITLE
refactor(notice): generate styles programmatically; improve contrast

### PR DESCRIPTION
**Pull request by dancormier**
> [!NOTE]
> This PRs is targeting https://github.com/StackExchange/Stacks/pull/1677

---

This PR refactors the code that generates styles for the `.s-notice`, `.s-banner`, and `.s-toast` components. This PR also:

- Unifies and improves child `code` element contrast
- Updates notice and toast documentation to generate examples programmatically
- Improves notice and toast documentation by showing examples that include `code` and `a.s-link.s-link__inherit` children where applicable
- Updates notice a11y and visual regression tests to include `code` and `a.s-link.s-link__inherit` children
- Updates baseline test images (hence the ~190 files changed)

### Child `code` contrast

@CGuindon I saw your comment (https://github.com/StackExchange/Stacks/pull/1677#issuecomment-1989445403) _right before_ I opened this PR 😂. I'm happy to wait to have this PR reviewed until you have the time to fully consider the design choices necessary to get this right. I know you're very busy these days, so no rush as far as I'm concerned.

Nevertheless, I took the liberty to make a few design choices for the `code` child element in this PR. These can be changed pretty easily, but I think I at least found a decent starting place. In this PR, all `code` elements within the notice component meet or exceed our accessibility contrast goals.

### Screenshots

<details>
<summary>Light mode</summary>

#### Notice base styles

image

#### Notice `important` variant styles

image

</details>

<details>
<summary>Dark mode</summary>

#### Notice base styles

image

#### Notice `important` variant styles

image

</details>

<details>
<summary>High contrast light mode</summary>

#### Notice base styles

image

#### Notice `important` variant styles

image

</details>

<details>
<summary>High contrast dark mode</summary>

#### Notice base styles

image

#### Notice `important` variant styles

image

</details>

### CODE REVIEW

**Comment by hamvocke** [comment]
I love it! Not much to say on the code itself from my end, but the a11y issue I care about here is massively improved 👍

**Review by CGuindon — APPROVED** [review]
Ship it!

File: lib/components/notice/notice.less [hunk 1]
```diff
@@ -1,109 +1,129 @@
+/**
+ * Generate color variables for a with a given color name
```
Comment (giamir) [inline lib/components/notice/notice.less]: small typo

**Review by giamir — APPROVED** [review]
Looks good to me too. Thanks @dancormier 🎉

**Pull request merged**